src/shogun/features/Features.cpp

```diff
@@ -265,6 +265,24 @@ void CFeatures::unset_property(EFeatureProperty p)
 	properties &= (properties | p) ^ p;
 }
 
+CFeatures* CFeatures::view(const SGVector<index_t>& subset)
+{
+	auto feats_view = this->duplicate();
```

Comment:
i would prefer a copy ctor here

Reply:
this is a virtual function that calls copy ctor of subclasses

Reply:
ah sorry I see now, totally justified!